### PR TITLE
Add Windows builds to test workflow

### LINKED ISSUE #200: Phylum CLI v1.2.0 fails to build on Windows with up-to-date rust installation
# Overview
Phylum does not package binary releases for Windows, forcing users that require Windows to build manually. Building the current release (v.1.20) fails using an up-to-date rust installation.

``` 
> cargo --version --verbose
cargo 1.59.0 (49d8809dc 2022-02-10)
release: 1.59.0
commit-hash: 49d8809dc2d3e6e0d5ec634fcf26d8e2aab67130
commit-date: 2022-02-10
host: x86_64-pc-windows-msvc
libgit2: 1.3.0 (sys:0.13.23 vendored)
libcurl: 7.80.0-DEV (sys:0.4.51+curl-7.80.0 vendored ssl:Schannel)
os: Windows 10.0.19044 (Windows 10 Pro) [64-bit]
```

```
> cargo install --path . --locked
   Compiling reqwest v0.11.6
error[E0433]: failed to resolve: could not find `unix` in `os`
 --> src\update.rs:5:14
  |
5 | use std::os::unix::fs::PermissionsExt;
  |              ^^^^ could not find `unix` in `os`

error[E0599]: no function or associated item named `from_mode` found for struct `Permissions` in the current scope
   --> src\update.rs:268:67
    |
268 |         fs::set_permissions(&installed_bin_path, fs::Permissions::from_mode(0o770))?;
    |                                                                   ^^^^^^^^^ function or associated item not found in `Permissions`

Some errors have detailed explanations: E0433, E0599.
For more information about an error, try `rustc --explain E0433`.
error: could not compile `phylum-cli` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `phylum-cli v1.2.0 (C:\Users\peter\projects\cli\lib)`, intermediate artifacts can be found at `C:\Users\peter\projects\cli\lib\target`

Caused by:
  build failed
```

# How To Reproduce
Steps to reproduce this behavior:
1. Install the current stable version of rust on Windows 10
2. Clone phylum-dev/cli
3. Checkout tag v1.2.0
4. Run `cargo install --path . --locked`
5. Observe error

# Expected Behavior
Current releases of Phylum CLI build correctly on Windows

### PR DESCRIPTION
**Pull request by kylewillmon**
# Overview

Use a matrix in the `test.yml` workflow to ensure that the CLI builds and all tests pass Linux, macos, and Windows. This also adds a scheduled run of the workflow to ensure that we are notified if the build begins to fail for external reasons (e.g., a new rust release). And this splits the self-update functions by platform in order to fix the build on Windows and close #200

## Checklist

- [x] Does this PR have an associated issue?
- [x] Have you ensured that you have met the expected acceptance criteria?
- [x] Have you created sufficient tests?
- [x] Have you updated all affected documentation?

### CODE REVIEW

**Review by maxrake — APPROVED** [review]
No showstoppers...but a few notes:
* I'm not yet familiar with Rust and so it might be worth having someone else who is also review
  * The proof is in the pudding, so to speak, and b/c the updated test actions pass...showing successful builds on Windows targets...I am marking as approved now
* It would be nice to start including Windows release artifacts now that they are available
  * Outside the scope of this issue/PR...but nice and likely worthy of a new issue

**Review by maxrake — DISMISSED** [review]
Brilliant addition of `test-matrix` and the follow on `test` job! That saves the manual repo settings changes when/if the test matrix changes. I'm going to ~steal~ re-use this pattern in the `phylum-ci` repo for the test matrix of Python versions. That will be huge since the supported list of Python versions is going to change at least once a year.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -9,16 +9,17 @@ on:
     branches:
       - master
       - development
+  # Run weekly on the development branch to make sure it always builds with the latest rust release
```
[the remark this answers is not in the document]
Comment (kylewillmon) [reply .github/workflows/test.yml:12]: My thinking was that the line above it says `development`, so I might as well. But I'm happy to change it.

File: cli/src/update/mod.rs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+//! Self-update functions
+//!
+//! Self-update is currently only supported on macos and Linux. All other platforms are unsupported
+//! and will display an error when `phylum update` is run.
```
[the remark this answers is not in the document]
Comment (kylewillmon) [reply cli/src/update/mod.rs]: Added issue #221 for Windows support.

I'm also taking the point here that I probably should not have written "_only_ supported on". Removing "only" makes this come across as more of a positive statement.

**Review by maxrake — APPROVED** [review]
LGTM

**Pull request merged**